neuraxle/steps/flow.py

```diff
@@ -0,0 +1,137 @@
+
+        :param data_container: data container to transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expaned_data_container(data_container)
+
+        expanded_data_container =  self.wrapped.handle_transform(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return expanded_data_container.reduce_dim()
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Send expanded data container to the wrapped handle_fit_transform method,
+        and returned the reduced transformed data container (back to it's orginal shape).
+
+        :param data_container: data container to fit_transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expaned_data_container(data_container)
+
+        self.wrapped, expanded_data_container = self.wrapped.handle_fit_transform(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return expanded_data_container.reduce_dim()
+
+    def handle_fit(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Send expanded data container to the wrapped handle_fit method,
+        and returned the reduced transformed data container (back to it's orginal shape).
+
+        :param data_container: data container to fit_transform
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: data container
+        :rtype: DataContainer
+        """
+        expanded_data_container = self._create_expaned_data_container(data_container)
+
+        self.wrapped, expanded_data_container = self.wrapped.handle_fit(
+            expanded_data_container,
+            context.push(self.wrapped)
+        )
+
+        return expanded_data_container.reduce_dim()
+
+    def _create_expaned_data_container(self, data_container) -> ExpandedDataContainer:
+        """
+        Create expanded data container.
+
+        :param data_container: data container to expand
+        :type data_container: DataContainer
+        :return: expanded data container
+        :rtype: ExpandedDataContainer
+        """
+        summary_hash = self.summary_hash(
+            data_container.current_ids,
+            self.hyperparams,
+            data_container.data_inputs
+        )
+
+        expanded_data_container = ExpandedDataContainer.create_from(data_container, summary_hash)
```

Comment:
I suggest to do that instead:
1. delete the summary_hash method from BaseStep
2. delete the summary_hash method from the hashers
3. add the summary_hash method in the base data container
4. here, hash like normally (like before) on each current_ids
5. then in the `ExpandedDataContainer.create_from`, just pass the data container such as `ExpandedDataContainer.create_from(data_container)`, and it will use `data_container.get_summary()` to compute a new single current_id. 

Why? 
- The reason we needed hashers was to sometimes consider values and hyperparameters, and sometimes not. 
- As long as the current_ids are updated, then we can always recompute the corresponding summary hash from within the data_container itself, without hashers.

Reply:
should a data container have a hasher instead ?

Reply:
I think not, for those reasons: 
1. the data container's summary hashing will probably never changes, and if it changes, it will be new types of data containers that overrides that behavior and not using more hashers
2. hashers are step data id hashers. Each different hashers should not need a different way to do a summary. Maybe what you are trying to do is like a `car-bicycle` that tries to do two things at once. I think current id hashers should not summarize.

Reply:
> We do not need many different ways to summarize many hashes into 1. 

Maybe later when using mini batches. Damn, I don't know.

Reply:
maybe, I'll call you thats too complicated to write.